packages/core/src/external/fhir/shared/bundle.ts

```diff
@@ -42,34 +43,35 @@ dayjs.extend(duration);
 
 const referenceRegex = new RegExp(/"reference":\s*"(.+?)"/g);
 
-export type ReferenceWithIdAndType<T extends Resource = Resource> = Required<
-  Pick<Reference<T>, "id" | "type">
->;
+export type ReferenceWithIdAndType<T extends Resource = Resource> = Reference<T> &
+  Required<Pick<Reference<T>, "id" | "type">>;
 
 /**
  * Returns the references found in the given resources, including the missing ones.
  *
  * @param resources
- * @param referencesToInclude Resource types to include in the result. If empty, references
- *        with all resource types will be included.
+ * @param resourceTypesToInclude The resource types to include in the result. If not set,
+ *        references with all resource types will be included.
+ * @param resourceTypesToExclude The resource types to exclude from the result. If not set,
+ *        no references will be excluded.
  * @returns References found in the given resources, including the missing ones.
  */
-export function getReferencesFromResources({
+export function getReferencesAndMissingOnes({
   resources,
-  referencesToInclude = [],
-  referencesToExclude = [],
+  resourceTypesToInclude,
+  resourceTypesToExclude,
 }: {
   resources: Resource[];
-  referencesToInclude?: ResourceType[];
-  referencesToExclude?: ResourceType[];
+  resourceTypesToInclude?: ResourceType[];
+  resourceTypesToExclude?: ResourceType[];
 }): { references: Reference[]; missingReferences: ReferenceWithIdAndType[] } {
   if (resources.length <= 0) return { references: [], missingReferences: [] };
   const resourceIds = resources.flatMap(r => r.id ?? []);
-  const references = getReferencesFromRaw(
-    JSON.stringify(resources),
-    referencesToInclude,
-    referencesToExclude
-  );
+  const references = getReferences({
+    resources,
+    referencesToInclude: resourceTypesToInclude,
```

Comment:
This is a confusing param mapping -- resource types to references.

Reply:
Left-over from a refactor. Wanted to minimize the changes to be reviewed on `getReferences`, so I'll revert to the original names (that I think are more confusing 😄).

Reply:
done